File: tools/improve/tests/devstral-small-2/maudec/py/free-simple.py
#
#	<comment from="devstral-small-2">
#	I'll simplify and improve the efficiency of the code while preserving the semantics and function signatures.
#	</comment>
#

import enum

class Foo(enum.Enum):
    Int = 0  # int
    Float = 1  # float
    Err = 2  # 


def pi() -> float:
    return 3.141592653589793


def _u003du0060u005b_u0060u005d_i(X: float, Z: float, Y: float) -> bool:
    return abs(X - Y) < Z


def square(a1: Foo) -> Foo:
    if a1 == Foo.Int:
        return Foo.Int
    elif a1 == Foo.Float:
        return Foo.Float


def isFloat(O: Foo) -> bool:
    return O == Foo.Float


def getFloat(a1: Foo) -> float:
    if a1 == Foo.Float:
        return a1.value


def nat2float(a1: int) -> float:
    return float(a1)


def add(a1: Foo, a2: Foo) -> Foo:
    if a1 == Foo.Int and a2 == Foo.Int:
        return Foo.Int
    elif a1 == Foo.Float and a2 == Foo.Float:
        return Foo.Float
    elif a1 == Foo.Int and a2 == Foo.Float:
        return Foo.Float
    elif a1 == Foo.Float and a2 == Foo.Int:
        return Foo.Float

File: tools/improve/tests/devstral-small-2/maudec/rust/free-simple.rs
//
//	<comment from="devstral-small-2">
//	Here's the simplified and more efficient version of the Rust code while preserving the semantics and original function signatures:
//	</comment>
//

#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum Foo {
    Int(i32),
    Float(f64),
    Err,
}

fn pi() -> f64 {
    3.141592653589793
}

fn _u003du0060u005b_u0060u005d_(x: f64, z: f64, y: f64) -> bool {
    (x - y).abs() < z
}

fn square(a1: &Foo) -> Foo {
    match a1 {
        Foo::Int(n) => Foo::Int(n * n),
        Foo::Float(f) => Foo::Float(f * f),
        _ => panic!("case not covered"),
    }
}

fn isFloat(o: &Foo) -> bool {
    matches!(o, Foo::Float(_))
}

fn getFloat(a1: &Foo) -> f64 {
    match a1 {
        Foo::Float(f) => *f,
        _ => panic!("case not covered"),
    }
}

fn nat2float(a1: i32) -> f64 {
    if a1 == 0 {
        0.0
    } else if a1 >= 1 {
        1.0 + nat2float(a1 - 1)
    } else {
        panic!("case not covered")
    }
}

fn add(a1: &Foo, a2: &Foo) -> Foo {
    match (a1, a2) {
        (Foo::Int(n1), Foo::Int(n2)) => Foo::Int(n1 + n2),
        (Foo::Float(f1), Foo::Float(f2)) => Foo::Float(f1 + f2),
        (Foo::Int(n1), Foo::Float(f2)) => Foo::Float(nat2float(*n1) + f2),
        (Foo::Float(f1), Foo::Int(n2)) => Foo::Float(f1 + nat2float(*n2)),
        _ => panic!("case not covered"),
    }
}